src/settings/NoticePolicy/components/EditSections/components/NoticesList/NoticesList.test.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import {
+  render,
+  screen,
+  within,
+} from '@testing-library/react';
+import fireEvent from '@testing-library/user-event';
+
+import '../../../../../../../test/jest/__mock__';
+
+import NoticesList from './NoticesList';
+import NoticeCard from '../NoticeCard';
+
+jest.mock('../NoticeCard', () => jest.fn(({
+  onRemoveNotice,
+  noticeIndex,
+}) => (
+  // eslint-disable-next-line jsx-a11y/click-events-have-key-events, jsx-a11y/no-static-element-interactions
+  <div
+    data-testid={`noticeCard${noticeIndex}`}
+    onClick={() => onRemoveNotice(noticeIndex)}
+  />
+)));
+
+describe('NoticesList', () => {
+  const labelIds = {
+    addNotice: 'ui-circulation.settings.noticePolicy.addNotice',
+  };
+  const testFieldsNames = [
+    'notice1',
+    'notice2',
+  ];
+  const testFieldsValue = [
+    {
+      value: 'value1',
+    },
+    {
+      value: 'value2',
+    },
+  ];
+  const sendEventTriggeringIds = ['sendEventTriggeringIds'];
+  const templates = [
+    {
+      value: 'templatesValue1',
+      label: 'templatesLabel1',
+    },
+  ];
+  const triggeringEvents = [
+    {
+      value: 'sendEventTriggeringIdsValue',
+      label: 'sendEventTriggeringIdsLabel',
+    },
+  ];
+  const testFields = {
+    push: jest.fn(),
+    remove: jest.fn(),
+    map: (callback) => testFieldsNames.map(callback),
+    value: testFieldsValue,
+  };
+  const sectionKey = 'feeFineNotices';
+  const policy = {
+    [sectionKey]: [
+      {
+        sendOptions: {
+          sendWhen: 'sendWhen1',
+        },
+      },
+      {
+        sendOptions: {
+          sendWhen: 'sendWhen2',
+        },
+      },
+    ],
+  };
+  const noticesListProps = {
+    getSendEvents : jest.fn(),
+    fields: testFields,
+    sectionKey,
+    policy,
+    sendEventTriggeringIds,
+    templates,
+    triggeringEvents,
+  };
+  const renderComponent = () => {
+    render(
+      <NoticesList
+        {...noticesListProps}
+      />
+    );
+  };
+
+  afterEach(() => {
+    NoticeCard.mockClear();
+    testFields.push.mockClear();
+    testFields.remove.mockClear();
+  });
+
+  describe(('notice cards'), () => {
+    beforeEach(() => {
+      renderComponent();
+    });
+
+    const generateTests = (testFieldName, noticeIndex) => {
+      describe((`notice card #${noticeIndex}`), () => {
```

Comment:
Do we really need `describe` with "...#0"? Or should it start from "1" for example? If it is, we can extract it in constant, and also use it in `toHaveBeenNthCalledWith` as call order number.

Reply:
Fixed